Core/test/qa-functional/src/org/sleuthkit/autopsy/modules/encryptiondetection/EncryptionDetectionTest.java

```diff
@@ -0,0 +1,198 @@
+
+            for (AbstractFile file : results) {
+                /*
+                 * Process only non-slack files.
+                 */
+                if (file.isFile() && !file.getType().equals(TskData.TSK_DB_FILES_TYPE_ENUM.SLACK)) {
+                    /*
+                     * Determine which assertions to use for the file based on
+                     * its name.
+                     */
+                    boolean fileProtected = file.getName().split("\\.")[0].endsWith("-protected");
+                    List<BlackboardArtifact> artifactsList = file.getAllArtifacts();
+                    if (fileProtected) {
+                        /*
+                         * Check that the protected file has one
+                         * TSK_ENCRYPTION_DETECTED artifact.
+                         */
+                        int artifactsListSize = artifactsList.size();
+                        String errorMessage = String.format("File '%s' (objId=%d) has %d artifacts, but 1 was expected", file.getName(), file.getId(), artifactsListSize);
+                        assertEquals(errorMessage, 1, artifactsListSize);
+
+                        String artifactTypeName = artifactsList.get(0).getArtifactTypeName();
+                        errorMessage = String.format("File '%s' (objId=%d) has an unexpected '%s' artifact.", file.getName(), file.getId(), artifactTypeName);
+                        assertEquals(errorMessage, BlackboardArtifact.ARTIFACT_TYPE.TSK_ENCRYPTION_DETECTED.toString(), artifactTypeName);
+                    } else {
+                        /*
+                         * Check that the unprotected file has no artifacts.
+                         */
+                        int artifactsListSize = artifactsList.size();
+                        String errorMessage = String.format("File '%s' (objId=%d) has %d artifacts, but none were expected", file.getName(), file.getId(), artifactsListSize);
+                        assertEquals(errorMessage, 0, artifactsListSize);
+                    }
+                }
+            }
+        } catch (NoCurrentCaseException | TskCoreException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        }
+    }
+
+    private void runIngestJob(List<Content> datasources, IngestModuleFactory factory) {
+        IngestModuleIngestJobSettings settings = factory.getDefaultIngestJobSettings();
+        IngestModuleTemplate template = new IngestModuleTemplate(factory, settings);
+        template.setEnabled(true);
+        ArrayList<IngestModuleTemplate> templates = new ArrayList<>();
+        templates.add(template);
+        IngestJobSettings ingestJobSettings = new IngestJobSettings(EncryptionDetectionTest.class.getCanonicalName(), IngestType.FILES_ONLY, templates);
+        try {
+            List<IngestModuleError> errs = IngestJobRunner.runIngestJob(datasources, ingestJobSettings);
+            assertEquals(0, errs.size());
```

Comment:
We should use the assertEquals which includes a message argument so we can log a failure reason instead of just the exceptions stack trace.